### PR TITLE
Chapter 10

### PR DESCRIPTION
**Pull request by SimonGross**
Chapter 10 answers

### CODE REVIEW

File: src/com/datacash/training/chapter10/Question7.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	private static final String INPUT_FILE_NAME = "src/com/datacash/training/chapter10/input.txt";
+	private static final String OUTPUT_FILE_NAME = "src/com/datacash/training/chapter10/output.txt";
+
+	public static void main( String[] args ) {
+		FileInputStream inputFile = null;
+		FileOutputStream outputFile = null;
+		try {
+			inputFile = new FileInputStream( INPUT_FILE_NAME );
+
+			List<Integer> output = new ArrayList<Integer>();
```
Comment (alexdeas) [inline src/com/datacash/training/chapter10/Question7.java:21]: What's with the List approach? Is it any faster or do you just like lists?

File: src/com/datacash/training/chapter10/answers.txt [hunk 2]
```diff
@@ -0,0 +1,38 @@
+1) The byte streams were originally the only I/O streams for Java, so when character streams were added byte streams had to be maintained for backwards compatability.
```
Comment (alexdeas) [inline src/com/datacash/training/chapter10/answers.txt:1]: That, and for times when you want to stream some bytes I guess.

File: src/com/datacash/training/chapter10/Question8.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import java.util.List;
+
+public class Question8 {
+
+	private static final String INPUT_FILE_NAME = "src/com/datacash/training/chapter10/input2.txt";
+	private static final String OUTPUT_FILE_NAME = "src/com/datacash/training/chapter10/output2.txt";
+
+	public static void main( String[] args ) {
+		try (FileReader inputFile = new FileReader( INPUT_FILE_NAME );
+		    FileWriter outputFile = new FileWriter( OUTPUT_FILE_NAME );)
```
Comment (alexdeas) [inline src/com/datacash/training/chapter10/Question8.java:17]: Superfluous semi-colon nooooooooooo

File: src/com/datacash/training/chapter10/Question7.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+			outputFile = new FileOutputStream( OUTPUT_FILE_NAME );
+			// Why doesn't this work? Because forEach doesn't provide exception handling!
+			//output.stream().forEach( outputFile::write );
+			for ( Integer i : output ) {
+				outputFile.write( i );
+			}
+
+		}
+		catch ( IOException ioe ) {
```
Comment (alexdeas) [inline src/com/datacash/training/chapter10/Question7.java:45]: Didn't you need to handle FileNotFoundException? I'm sure Eclipse moaned about it while I was writing my answer.

Comment (SimonGross) [inline src/com/datacash/training/chapter10/Question7.java:45]: Oddly enough Eclipse didn't complain about it for me.

File: src/com/datacash/training/chapter10/Question8.java (hunk 3 above)
Comment (SimonGross) [inline src/com/datacash/training/chapter10/Question8.java:17]: :sob:

File: src/com/datacash/training/chapter10/Question7.java (hunk 1 above)
Comment (SimonGross) [inline src/com/datacash/training/chapter10/Question7.java:21]: I was originally planning to use some Java 8 features, but that didn't work out so well as I found out further down.

**Pull request closed without merging**